### PR TITLE
stop rewrite DeletedAt for every call

### PR DESCRIPTION
**Pull request by lukas016**
# Proposed Changes

DeletdeAt in api.Machine was rewrite for every iri call DeleteMachine. It created conflicts with other components which trying update api.Machine object and information when deleteMachine call was called first time was lost.

### CODE REVIEW

File: pkg/host/store.go [hunk 1]
```diff
@@ -165,12 +165,14 @@ func (s *Store[E]) Delete(_ context.Context, id string) error {
...
+		obj.IncrementResourceVersion()
 
-	if _, err := s.set(obj); err != nil {
-		return fmt.Errorf("failed to set object metadata: %w", err)
+		if _, err := s.set(obj); err != nil {
+			return fmt.Errorf("failed to set object metadata: %w", err)
+		}
 	}
 
 	s.enqueue(store.WatchEvent[E]{
```
Comment (so-sahu) [inline pkg/host/store.go:178]: If a second delete request is sent, nothing happens in this code. Do we still need to trigger the delete event in this scenario?

Comment (lukas016) [reply pkg/host/store.go:178]: i don't know, so i kept previous logic. Minimally it will still trigger reconcile loop.

Comment (so-sahu) [reply pkg/host/store.go:178]: It doesn't seem necessary to trigger the reconcile loop if there are no changes.

Comment (lukas016) [reply pkg/host/store.go:178]: it is the design problem, you don't retrigger reconcile loop only but you notified every goroutines which registered watch function. So i kept this logic. @lukasfrank you will know better. I agree with @so-sahu and it isn't needed probably but i wanted risk side effects with designed change.

Comment (lukas016) [reply pkg/host/store.go:178]: you have right, i checked Update function and when object doesn't contain changes, it doesn't send event. I will change it.

Comment (lukas016) [reply pkg/host/store.go]: fixed

**Review by so-sahu — APPROVED** [review]
LGTM

**Review by hardikdr — APPROVED** [review]

**Pull request merged**